### PR TITLE
[expo-updates] Make extra header processing code consistent between manifests and assets

### PR DESCRIPTION
**Pull request by wschurman**
# Why

The code for manifest extra headers has always had methods for coercing into strings. The asset extra header code diverged from this (from the time it was added).

# How

Make extra header processing code the same for all requests.

# Test Plan

Run tests.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: packages/expo-updates/android/src/androidTest/java/expo/modules/updates/loader/FileDownloaderTest.kt [hunk 1]
```diff
@@ -123,6 +123,34 @@ class FileDownloaderTest {
...
+      put("expo-number", 47.5)
+      put("expo-boolean", true)
+    }
+
+    val assetEntity = AssetEntity("test", "jpg").apply {
+      url = Uri.parse("https://example.com")
+      extraRequestHeaders = extraHeaders
+    }
+
+    // assetRequestHeaders should not be able to override preset headers
```
Comment (esamelson) [inline packages/expo-updates/android/src/androidTest/java/expo/modules/updates/loader/FileDownloaderTest.kt:147]:
```suggestion
    // assetRequestHeaders should have their values coerced to strings
```
or something to that effect

**Review by esamelson — APPROVED** [review]
LGTM. Thanks for the fix.

File: packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt [hunk 2]
```diff
@@ -472,20 +472,25 @@ open class FileDownloader(private val client: OkHttpClient) {
       throw IOException("No compatible manifest found. SDK Versions supported: " + configuration.sdkVersion + " Provided manifestString: " + manifestString)
     }
 
+    private fun Request.Builder.addHeadersFromJSONObject(headers: JSONObject?): Request.Builder {
+      if (headers == null) {
+        return this
+      }
+
+      headers.keys().asSequence().forEach { key ->
+        header(key, headers.require<Any>(key).toString())
```
Comment (esamelson) [inline packages/expo-updates/android/src/main/java/expo/modules/updates/loader/FileDownloader.kt:481]: oh, just now remembering that @douglowder made a good point about null values in JSON -- I think that in the iOS implementation that will at least not crash (though might send a weird value) but will that be handled correctly here?

File: packages/expo-updates/android/src/androidTest/java/expo/modules/updates/loader/FileDownloaderTest.kt [hunk 3]
```diff
@@ -123,6 +123,34 @@ class FileDownloaderTest {
...
+      "updateUrl" to Uri.parse("https://u.expo.dev/00000000-0000-0000-0000-000000000000"),
+      "runtimeVersion" to "1.0",
+    )
+
+    val config = UpdatesConfiguration(null, configMap)
+
+    val extraHeaders = JSONObject().apply {
+      put("expo-string", "test")
+      put("expo-number", 47.5)
+      put("expo-boolean", true)
```
Comment (douglowder) [inline packages/expo-updates/android/src/androidTest/java/expo/modules/updates/loader/FileDownloaderTest.kt:139]:
```suggestion
      put("expo-boolean", true)
      put("expo-null", JSONObject.NULL)
```

Comment (douglowder) [inline packages/expo-updates/android/src/androidTest/java/expo/modules/updates/loader/FileDownloaderTest.kt:139]:
```suggestion
      put("expo-boolean", true)
      put("expo-null", JSONObject.NULL)
```

**Review by douglowder — APPROVED** [review]
Looks good overall, but I do suggest that if null values are supported, that we add some test for that.

**Comment by wschurman** [comment]
Excellent suggestions on the null case handling. The android implementation serialized to `"null"` while the iOS implementation serialized to `"<null>"`. I changed the iOS one to match android.

**Pull request merged**